roles/openshift_master/handlers/main.yml

```diff
@@ -1,4 +1,4 @@
 ---
 - name: restart master
   service: name={{ openshift.common.service_type }}-master state=restarted
-  when: not openshift_master_ha | bool
+  when: not (openshift_master_ha | bool or skip_master_restart | default(false))
```

Comment:
yikes.

The other bit I'm struggling with is that something in me prefers to flip the logic around a bit and rename `skip_master_restart` to `master_restart_needed`.  Combining `not`, `or` and `skip` gets me worried about maintenance.  Someone might add on more logic or move the parentheses and mess it all up.  It definitely feels dense.  I'll try to look over this again tomorrow.